src/utils.jl

```diff
@@ -56,11 +58,51 @@ end
 
 # family of projection functions
 
-immutable Proj{field}; end
+immutable Proj{F}
+    f::F
```

Comment:
Okay, I can separate this out into a `ProjFunction` type. Right now I use `Val{f}` for the case you mentioned above.

`(p::Proj)(c::Columns)` can be removed.

> should be a method of broadcast now.

And `map` too right?